dialog_login.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/* Copyright (c) 1998-2009 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+* login script for ilias
+*
+* @author Sascha Hofmann <saschahofmann@gmx.de>
+* @author Peter Gabriel <pgabriel@databay.de>
+* @version $Id$
+*
+* @package ilias-layout
+*/
+
+include('./maintenance.inc.php');
+
+// jump to setup if ILIAS3 is not installed
+if (!file_exists(getcwd()."/ilias.ini.php")) {
+	header("Location: ./setup/setup.php");
+	exit();
+}
+
+// start correct client
+// if no client_id is given, default client is loaded (in class.ilias.php)
+if (isset($_GET["client_id"])) {
+	$cookie_domain = $_SERVER['SERVER_NAME'];
+	$cookie_path = dirname($_SERVER['PHP_SELF']);
+
+	/* if ilias is called directly within the docroot $cookie_path
+	is set to '/' expecting on servers running under windows..
+	here it is set to '\'.
+	in both cases a further '/' won't be appended due to the following regex
+	*/
+	$cookie_path .= (!preg_match("/[\/|\\\\]$/", $cookie_path)) ? "/" : "";
+
+	if ($cookie_path == "\\") {
+		$cookie_path = '/';
+	}
+
+	$cookie_domain = ''; // Temporary Fix
+
+	setcookie("ilClientId", $_GET["client_id"], 0, $cookie_path, $cookie_domain);
+
+	$_COOKIE["ilClientId"] = $_GET["client_id"];
+}
+
+if((isset($_GET["cmd"]) && $_GET["cmd"] != "post")
+	|| (isset($_GET["cmdClass"]) && $_GET["cmdClass"] != "ilstartupgui")
+) {
+	$url = str_replace("dialog_login.php", "ilias.php", $_SERVER["REQUEST_URI"]);
+	header("location: $url");
+}
+
+require_once "Services/Style/classes/class.ilStyleDefinition.php";
```

Comment:
pls be consistent in styling require_once in the whole file even it is generali.

Reply:
Why should i require an element, if i get redirected before and there is no use?
And remember this is no class, just a script.